File: src/main/java/kitchenpos/menus/tobe/menugroup/domain/repository/MenuGroupRepositoryV2.java
package kitchenpos.menus.tobe.menugroup.domain.repository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;
import kitchenpos.menus.tobe.menugroup.domain.model.MenuGroupV2;

public interface MenuGroupRepositoryV2 {
    MenuGroupV2 save(MenuGroupV2 menuGroup);

    Optional<MenuGroupV2> findById(UUID id);

    List<MenuGroupV2> findAll();
}

File: src/main/kotlin/kitchenpos/tobe/menu/domain/repository/MenuGroupRepositoryV2.kt
package kitchenpos.tobe.menu.domain.repository

import kitchenpos.tobe.menu.domain.entity.MenuGroupV2
import java.util.UUID

interface MenuGroupRepositoryV2 {
    fun findMenuGroupById(id: UUID): MenuGroupV2?

    fun save(menuGroup: MenuGroupV2): MenuGroupV2
}
